### PR TITLE
Add support for derive endpoint in identity and authorize derive keys

### PR DESCRIPTION
**Pull request by lazynina**
also rename `node` to `Node`

### CODE REVIEW

File: apps/developer-hub/src/chapters/ChapterHelper/IdentityChapters.tsx [hunk 1]
```diff
@@ -116,4 +119,74 @@ export const identityChapter = {
       );
     },
   },
+  IDENTITY_DERIVE: {
```
[the remark this answers is not in the document]
Comment (lazynina) [reply apps/developer-hub/src/chapters/ChapterHelper/IdentityChapters.tsx:122]: *however the key cannot sign transactions subject to the spending limits until the user signs and submits the transaction. Due to the sensitive nature of this transaction type, we will not be signing and submitting it in the hub. In order to sign and submit the transaction, the user would want to flip the `broadcast` argument to true

File: babel.config.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "babelrcRoots": ["*"],
-  "presets": ["@babel/preset-flow"],
+  "presets": ["@babel/preset-flow", "@babel/preset-typescript"],
```
[the remark this answers is not in the document]
Comment (lazynina) [reply babel.config.json:3]: Enum type definitions weren't happy until I added this. Cause typescript errors without it.

File: libs/deso-protocol/src/lib/identity/Identity.ts [hunk 3]
```diff
@@ -85,14 +98,31 @@ export class Identity {
...
-    const prompt = requestLogout(publicKey);
+    const prompt = requestLogout(publicKey, this.getUri());
     const successful = await iFrameHandler({
       iFrameMethod: 'logout',
       data: { prompt },
     });
     return successful;
   }
 
+  public async derive(params: IdentityDeriveParams): Promise<DerivedPrivateUserInfo> {
```
[the remark this answers is not in the document]
Comment (lazynina) [reply libs/deso-protocol/src/lib/identity/Identity.ts:109]: These structs are from identity, not backend.

File: libs/deso-protocol/src/lib/identity/WindowPrompts.ts [hunk 4]
```diff
@@ -1,26 +1,41 @@
...
     null as unknown as any,
     'toolbar=no, width=800, height=1000, top=0, left=0'
   ) as Window;
   return prompt;
 };
+
+export const requestDerive = (params: IdentityDeriveQueryParams, uri: string) => {
+  const queryParams = Object.entries(params || {}).
+    filter(([_, value]) => value !== null && value !== undefined).
+    map(([key, value]) => `${key}=${value.toString()}`);
```
[the remark this answers is not in the document]
Comment (lazynina) [reply libs/deso-protocol/src/lib/identity/WindowPrompts.ts:33]: `toString` is defined on boolean primitives! but I'm not sure if there would be an issue with the `TransactionSpendingLimitResponse` without it. can try and get back to you.

File: libs/deso-protocol/src/lib/identity/WindowHandler.ts [hunk 5]
```diff
@@ -80,9 +80,13 @@ export const handlers = async (
...
   if (info.iFrameMethod === 'encrypt') {
     if (event.data.payload.encryptedMessage) {
-      console.log(event.data);
       info.data.resolve(event.data.payload.encryptedMessage);
       window.removeEventListener('message', windowHandler);
     }
   }
+  if (info.iFrameMethod === 'derive' && event.data.method === 'derive') {
+    info.data.prompt?.close();
+    info.data.resolve(event.data.payload as DerivedPrivateUserInfo);
```
[the remark this answers is not in the document]
Comment (lazynina) [reply libs/deso-protocol/src/lib/identity/WindowHandler.ts:89]: it just makes it clear what the type is?

File: babel.config.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (DeSoDog) [reply babel.config.json:3]: Got it

File: apps/developer-hub/src/chapters/ChapterHelper/UserChapter.tsx [hunk 6]
```diff
@@ -341,6 +343,64 @@ export const userChapter = {
       );
     },
   },
+  AUTHORIZE_DERIVED_KEY: {
```
[the remark this answers is not in the document]
Comment (lazynina) [reply apps/developer-hub/src/chapters/ChapterHelper/UserChapter.tsx:346]: This call does both - it generates the derived key by calling `Identity.derive` and then hits the backend to construct the transaction based on the output from Identity.

File: libs/deso-protocol/src/lib/identity/WindowPrompts.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lazynina) [reply libs/deso-protocol/src/lib/identity/WindowPrompts.ts]: yeah works without .toString. resolved in latest commit.

File: libs/deso-protocol/src/lib/identity/WindowHandler.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DeSoDog) [reply libs/deso-protocol/src/lib/identity/WindowHandler.ts:89]: sg

File: dist/libs/deso-protocol/src/index.d.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ import { Node } from './lib/Node/Node';
 import { Wallet } from './lib/wallet/Wallet';
 import { Transactions } from './lib/transaction/Transaction';
 export declare class Deso {
-    constructor();
+    constructor(nodeUri?: string, identityConfig?: IdentityConfig);
```
Comment (DeSoDog) [inline dist/libs/deso-protocol/src/index.d.ts:14]: Lets makes this a single param also so 
export interface DesoConfig {nodeUri?: string, identityConfig: IdentityConfig}

File: libs/deso-protocol/src/index.ts [hunk 8]
```diff
@@ -14,11 +14,14 @@ import { Tutorial } from './lib/tutorial/Tutorial';
 import { Wallet } from './lib/wallet/Wallet';
 import { Transactions } from './lib/transaction/Transaction';
 export class Deso {
-  constructor() {
+  constructor(nodeUri?: string, identityConfig?: IdentityConfig) {
```
Comment (DeSoDog) [inline libs/deso-protocol/src/index.ts:17]: See my comment above on adding a DesoConfig here, accidently wrote it for the dist file

**Pull request merged**